src/app/product-details-page/components/product-details-page/product-details-page.component.html

```diff
@@ -1 +1,2 @@
 <app-product-description [product]="productInfo"></app-product-description>
+<app-related-products [products]="product?.relatedProducts"></app-related-products>
```

Comment:
it's better not to instantiate related-product component at all if there are no products
```html
<app-related-products *ngIf="product?.relatedProducts" [products]="product.relatedProducts"></app-related-products>
```